### DISCUSSION TITLE
How i check invalid WhatsApp Number before send message

### DISCUSSION THREAD

**Discussion by spacesoft321** [WhiskeySockets/Baileys#336] [General]
hello sir 

i use it , but its send all number , if is WhatsApp number or not

same problem in Waziper  code 

please add invalid whatsapp number function also

**Comment by dotKa** [comment]
Already has
```
const id = '123456'
const [result] = await sock.onWhatsApp(id)
if (result.exists) console.log (`${id} exists on WhatsApp, as jid: ${result.jid}`)
```

**Reply by spacesoft321** [reply]
hi
thanks for reply 

**i want is_whatsapp_number or not** 

i use waziper code , my website url : nvm.fastwsapi.com or my.fastwsapi.com

i try send message only valid whatsapp number 

if i send **918878596325** message send and delivered

but when i testing **92**8878596325 and **93**8878596325 and **94**8878596325 ( i just change **countries code** ) then also message send , 

i know its **all invalid number** but message send by waziper api code 

how i fix this error using **waziper api**

**Comment by spacesoft321** [comment]
how i add new function **Check is whatsapp number or not**  ,before send whatsapp message using **waziper api** 

if i send 918878596325 message send and delivered

but when i testing 928878596325 and 938878596325 and 948878596325 ( i just change countries code ) then also message send ,

i know its all invalid number but message send by waziper api code

how i fix this error using waziper api


    public function send(){
        $team = self::get_team();
        $team_id = $team->id;
        $access_token = $team->ids;
        $instance_id = self::get_instance_id();
        $type = post("type");
        $message = post("message");
        $filename = post("filename");
        $media_url = post("media_url");
        $number = post("number");

        $session = db_get("*", TB_WHATSAPP_SESSIONS, ["team_id" => $team_id, "instance_id" => $instance_id]);

        if( !$session ){
            ms([ "status" => "error", "message" => __("Instance ID Invalidated") ]);
        }

        if( $session->status == 0 ){
            ms([ "status" => "error", "message" => __("This instance ID has not been activated yet") ]);
        }

        $account = db_get("*", TB_ACCOUNTS, ["team_id" => $team_id, "token" => $instance_id]);

        if(!$account){
            ms([ "status" => "error", "message" => __("Account does not exist") ]);
        }

        if( $account->status == 0 ){
            ms([ "status" => "error", "message" => "This WhatsApp account relogin required" ]);
        }

        if($number == ""){
            ms([ "status" => "error", "message" => __("Phone number is required") ]);
        }

        if($media_url == "" && $message == ""){
            ms([ "status" => "error", "message" => __("Please enter media url or message") ]);
        }

        $response = wa_post_curl("send_message", [
            "instance_id" => $instance_id, 
            "access_token" => $access_token
        ], [
            "media_url" => $media_url,
            "chat_id" => $number."@c.us",
            "caption" => $message,
            "filename" => $filename
        ] );

        pr($response,1);

        ms((array)$response);
    }
